backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -256,6 +257,7 @@ public AnalysisConfiguration getAnalysisConfigForEngine(String configScopeId, Li
     analysisProperties.putAll(inferredAnalysisProperties);
     var baseDir = fileSystemService.getBaseDir(configScopeId);
     var actualBaseDir = baseDir == null ? findCommonPrefix(filePathsToAnalyze) : baseDir;
+    filePathsToAnalyze = GitUtils.filterOutIgnoredFiles(actualBaseDir, filePathsToAnalyze);
```

Comment:
It would be better to move this filtering to the `toInputFiles` method since serverFileExclusion filtering is already there. Better to keep them together

```java
  private List<ClientInputFile> toInputFiles(String configScopeId, List<URI> fileUrisToAnalyze) {
    return fileUrisToAnalyze.stream()
      .filter(not(fileExclusionService::isExcluded))
      .map(uri -> toInputFile(configScopeId, uri))
      .filter(Objects::nonNull)
      .collect(toList());
  }
 ```

Reply:
Quick question. Is this filtering also necessary for auto-triggered analysis? I think yes. If so, we should filter out the files in `triggerAnalysis` method in `AnalysisService`

Reply:
the `triggerAnalysis` method will eventually call the `analyze()`  which in turn will call `getAnalysisConfigForEngine()` and the `toInputFiles()` - so I think the original suggestion of moving it to `toInputFiles()` is good?

Reply:
Ah, yes. I missed that.

Reply:
Moved to `toInputFiles` in the last commit (34ae30b02d40dedea6c77724771076313aa9ef49)